subgraphs/collectible-market/mappings/index.ts

```diff
@@ -0,0 +1,263 @@
+  let collection = Collection.load(event.params.collection.toHex());
+  if (collection == null) {
+    collection = new Collection(event.params.collection.toHex());
+    collection.name = fetchCollectionName(event.params.collection);
+    collection.symbol = fetchCollectionSymbol(event.params.collection);
+    collection.active = true;
+    collection.totalTrades = ZERO_BI;
+    collection.totalVolumeBNB = ZERO_BD;
+    collection.numberTokensListed = ZERO_BI;
+    collection.creatorAddress = event.params.creator.toHex();
+    collection.tradingFee = toBigDecimal(event.params.tradingFee, 2);
+    collection.creatorFee = toBigDecimal(event.params.creatorFee, 2);
+    collection.whitelistChecker = event.params.whitelistChecker.toHex();
+  } else {
+    // Collection has existed, was closed, but is re-listed
+    collection.active = true;
+    collection.creatorAddress = event.params.creator.toHex();
+    collection.tradingFee = toBigDecimal(event.params.tradingFee, 2);
+    collection.creatorFee = toBigDecimal(event.params.creatorFee, 2);
+    collection.whitelistChecker = event.params.whitelistChecker.toHex();
+  }
+  collection.save();
+}
+
+export function handleCollectionClose(event: CollectionClose): void {
+  let collection = Collection.load(event.params.collection.toHex());
+  collection.active = false;
+  collection.save();
+}
+
+export function handleCollectionUpdate(event: CollectionUpdate): void {
+  let collection = Collection.load(event.params.collection.toHex());
+  collection.creatorAddress = event.params.creator.toHex();
+  collection.tradingFee = toBigDecimal(event.params.tradingFee, 2);
+  collection.creatorFee = toBigDecimal(event.params.creatorFee, 2);
+  collection.whitelistChecker = event.params.whitelistChecker.toHex();
+  collection.save();
+}
+
+/**
+ * ASK ORDERS
+ */
+
+export function handleAskNew(event: AskNew): void {
+  // 1. User
+  let user = User.load(event.params.seller.toHex());
+  if (user == null) {
+    user = new User(event.params.seller.toHex());
+    user.numberTokensListed = ONE_BI;
+    user.numberTokensPurchased = ZERO_BI;
+    user.numberTokensSold = ZERO_BI;
+    user.totalVolumeInBNBTokensPurchased = ZERO_BD;
+    user.totalVolumeInBNBTokensSold = ZERO_BD;
+    user.totalFeesCollectedInBNB = ZERO_BD;
+    user.averageTokenPriceInBNBPurchased = ZERO_BD;
+    user.averageTokenPriceInBNBSold = ZERO_BD;
+  } else {
+    user.numberTokensListed = user.numberTokensListed.plus(ONE_BI);
+  }
+
+  // 2. Collection
+  let collection = Collection.load(event.params.collection.toHex());
+  collection.numberTokensListed = collection.numberTokensListed.plus(ONE_BI);
+
+  // 3. Token
+  let tokenConcatId = event.params.collection.toString() + "-" + event.params.tokenId.toString();
+  let token = NFT.load(tokenConcatId);
+
+  if (token == null) {
+    token = new NFT(tokenConcatId);
+    token.tokenId = event.params.tokenId;
+    token.collection = event.params.collection.toHex();
```

Comment:
```suggestion
    token.collection = collection.id;
```
Please, reference the Entity ID previously loaded directly (both works, but it's better to ensure we know it's a relationship at first sight).